### PR TITLE
Add a credentials file path to the connector config

### PR DESCRIPTION
**Pull request by yuriy-tolochkevych-exa**

### CODE REVIEW

File: kafka-connector/README.md [hunk 1]
```diff
@@ -89,6 +85,7 @@ Connector supports the following configs:
 | kafka.key.attribute | String | null | The Cloud Pub/Sub message attribute to use as a key for messages published to Kafka. |
 | kafka.partition.count | Integer | 1 | The number of Kafka partitions for the Kafka topic in which messages will be published to. NOTE: this parameter is ignored if partition scheme is "kafka_partitioner".|
 | kafka.partition.scheme | round_robin, hash_key, hash_value, kafka_partitioner | round_robin | The scheme for assigning a message to a partition in Kafka. The scheme "round_robin" assigns partitions in a round robin fashion, while the schemes "hash_key" and "hash_value" find the partition by hashing the message key and message value respectively. "kafka_partitioner" scheme delegates partitioning logic to kafka producer, which by default detects number of partitions automatically and performs either murmur hash based partition mapping or round robin depending on whether message key is provided or not.|
+|gcp.credentials.file.path| String | REQUIRED (No default) | The file path, which stores GCP credentials.|
```
Comment (kamalaboulhosn) [inline kafka-connector/README.md]: Please make this an optional parameter. If it is not set, then it should create the channel the same way it did before.

**Comment by yuriy-tolochkevych-exa** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here (e.g. `I signed it!`) and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google.com/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).

I signed it!

**Comment by yuriy-tolochkevych-exa** [comment]
@kamalaboulhosn please check updates

File: kafka-connector/src/main/java/com/google/pubsub/kafka/common/ConnectorUtils.java [hunk 2]
```diff
@@ -48,16 +49,18 @@
...
-  public static Channel getChannel() throws IOException {
+  public static Channel getChannel(String credKeyPath) throws IOException {
     ManagedChannel channelImpl =
         NettyChannelBuilder.forAddress(ENDPOINT, 443)
             .negotiationType(NegotiationType.TLS)
             // Maximum Pub/Sub message size is 10MB.
             .maxInboundMessageSize(10 * 1024 * 1024)
             .build();
+    GoogleCredentials googleCredentials = credKeyPath == null ? 
+        GoogleCredentials.getApplicationDefault() : GoogleCredentials.fromStream(new FileInputStream(credKeyPath));
```
[the remark this answers is not in the document]
Comment (kamalaboulhosn) [reply kafka-connector/src/main/java/com/google/pubsub/kafka/common/ConnectorUtils.java]: Sorry, nevermind.

File: kafka-connector/src/main/java/com/google/pubsub/kafka/source/CloudPubSubGRPCSubscriber.java [hunk 3]
```diff
@@ -39,9 +39,11 @@
   private long nextSubscriberResetTime = 0;
   private SubscriberFutureStub subscriber;
   private Random rand = new Random(System.currentTimeMillis());
+  private String gcpCredentialsFilePath;
 
-  CloudPubSubGRPCSubscriber() {
+  CloudPubSubGRPCSubscriber(String gcpCredentialsFilePath) {
```
Comment (kamalaboulhosn) [inline kafka-connector/src/main/java/com/google/pubsub/kafka/source/CloudPubSubGRPCSubscriber.java]: We should make the same changes on the publisher side so one can specify credentials for the sink connector, too.

**Comment by yuriy-tolochkevych-exa** [comment]
@kamalaboulhosn Done, please see updates

**Comment by yuriy-tolochkevych-exa** [comment]
Thanks for review.  How long it will take  to be merged?

**Comment by kamalaboulhosn** [comment]
It is now merged.

**Pull request merged**